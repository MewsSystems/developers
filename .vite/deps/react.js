import {
  require_react
} from "./chunk-32E4H3EV.js";
import "./chunk-G3PMV62Z.js";
export default require_react();
//# sourceMappingURL=react.js.map
