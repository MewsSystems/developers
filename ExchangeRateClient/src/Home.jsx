// @flow
import React, { Component } from 'react';
import { List } from 'immutable';
import { connect } from 'react-redux';
import styled from 'styled-components';

import type { CurrencyPairRate } from './types';
import type { RequestStatus, CurrencyPair } from './reducer';
import { getCurrencyPairList, getCurrencyPairRates } from './selectors';
import { fetchConfig, fetchRates } from './actions';
import CurrencyPairsFilter from './CurrencyPairsFilter';
import RateList from './RateList';

type Props = {
  configRequestStatus: RequestStatus,
  currencyPairs: List<CurrencyPair>,
  rates: List<CurrencyPairRate>,
  fetchConfig: () => void,
  fetchRates: (interval: number, ids: Array<string>) => void,
};

type State = {};

const Container = styled.div`
  height: 100vh;
  width: 100vw;
  padding: 1rem 2rem;
`;

const Heading = styled.h1`
  margin: 0 0 2rem;
  font-size: 2rem;
`;

const StyledFilter = styled(CurrencyPairsFilter)`
  margin: 0 0 1rem;
`;

const StyledRateList = styled(RateList)`
  margin: 0 0 1rem;
`;

class Home extends Component<Props, State> {
  static defaultProps = {};

  state = {};

  componentDidMount() {
    this.props.fetchConfig();
  }

  render() {
    const { configRequestStatus, currencyPairs, rates } = this.props;

    return (
      <Container>
        <Heading>Exchange Rate App</Heading>
        {configRequestStatus === 'success' ? (
          <>
            <StyledFilter currencyPairs={currencyPairs} />
            <StyledRateList rates={rates} />
          </>
        ) : (
          <div>Loading...</div>
        )}
      </Container>
    );
  }
}

const mapStateToProps = state => ({
  configRequestStatus: state.getIn(['app', 'configRequestStatus']),
  currencyPairs: getCurrencyPairList(state),
  rates: getCurrencyPairRates(state),
});

const mapDispatchToProps = { fetchConfig, fetchRates };

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Home);
