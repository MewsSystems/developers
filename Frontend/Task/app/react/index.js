import React from 'react';
import { render } from 'react-dom';
import '@babel/polyfill';

import configureStore from './configurateStore';
import RootComponent from './components/RootComponent';

const initialState = window.__INITIAL_STATE__;
const store = configureStore({ initialState });
const rootElement = document.getElementById('reactRoot');

if (process.env.NODE_ENV !== 'production') {
  window.store = store;
}

render(
<RootComponent store={store} />,
rootElement
);

if (module.hot) {
  module.hot.accept('./components/RootComponent', () => {
    const NextRootComponent = require('./components/RootComponent').default;
    render(
      <NextRootComponent store={store}/>,
      rootElement
    );
  });
}
