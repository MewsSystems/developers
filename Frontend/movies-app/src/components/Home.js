import React, { useState, useEffect } from 'react';
import { useHistory, useLocation } from 'react-router-dom';
import { useSelector, useDispatch } from 'react-redux';
import URLSearchParams from 'query-string';

import * as actions from './../data/state/actions';
import useDebounce from './../hooks/useDebounce';
import MovieList from './MovieList';
import SearchBox from './SearchBox';
import Pagination from './Pagination';
import { Section, Row, Paragraph } from './styled';

const Home = () => {
  const { search } = useLocation();
  const params = URLSearchParams.parse(search);
  const pageIndex = params['page'] || 1;

  const history = useHistory();

  const dispatch = useDispatch();
  const movies = useSelector(state => state.movies.items);
  const currentPage = useSelector(state => state.movies.currentPage);
  const totalPages = useSelector(state => state.movies.totalPages);
  const [searchTerm, setSearchTerm] = useState('');
  const debouncedSearchTerm = useDebounce(searchTerm, 500);

  useEffect(() => {
    dispatch(actions.LoadMovies(debouncedSearchTerm, pageIndex));
  }, [debouncedSearchTerm, dispatch, pageIndex]);

  const pageChanged = newPage => {
    window.scrollTo(0, 0);
    history.push(`/?page=${newPage}`);
  };

  return (
    <>
      <Section>
        <SearchBox
          searchTerm={searchTerm}
          placeHolder="Search for movies"
          onSearchTermChanged={e => setSearchTerm(e.target.value)}
        />
      </Section>
      <Section>
        <Row align="right">
          <Pagination currentPage={currentPage} onPageChanged={pageChanged} totalPages={totalPages} />
        </Row>
        <MovieList movies={movies} empty={<Paragraph bold>No matches have been found.</Paragraph>} />
        <Row align="right">
          <Pagination currentPage={currentPage} onPageChanged={pageChanged} totalPages={totalPages} />
        </Row>
      </Section>
    </>
  );
};

export default Home;
