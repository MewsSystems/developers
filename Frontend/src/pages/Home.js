import React, { Fragment } from 'react';
import { connect } from 'react-redux';
import Search from '../components/movies/Search';
import Movies from '../components/movies/Movies';
import Pagination from '../components/movies/Pagination';
import { searchMovies, setSearchTerm } from '../actions/movieActions';

const Home = ({ searchMovies, setSearchTerm, movie: { searchTerm } }) => {

  const handleSearchChange = query => {
    setSearchTerm(query);
    searchMovies(query);
  };

  const handleClick = number => {
    searchMovies(searchTerm, number);
  };

  return (
    <Fragment>
      <div className="container">
        <Search change={handleSearchChange} />
        <Movies />
        <Pagination click={handleClick} />
      </div>
    </Fragment>
  )
};

const mapStateToProps = state => ({
  movie: state.movie
});

export default connect(mapStateToProps, { searchMovies, setSearchTerm })(Home);
