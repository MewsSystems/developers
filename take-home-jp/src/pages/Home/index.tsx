import React, { useState } from 'react';

import Input from '../../components/Input';
import ErrorLabel from '../../components/ErrorLabel';
import Pagination from '../../components/Pagination';
import MovieCard from '../../components/MovieCard';
import IntroScreen from '../../components/IntroScreen';
import useSearchMovies from '../../api/useSearchMovies';
import NoResultsSceen from '../../components/NoResultsScreen';
import { Movie } from '../../components/MovieCard/types';
import Container from '@mui/material/Container';
import Grid from '@mui/material/Grid';
import CircularProgress from '@mui/material/CircularProgress';
import Box from '@mui/material/Box';

const Homepage = () => {
  const {
    data,
    isLoading,
    error,
    search: searchMoviesByText,
    totalPages,
  } = useSearchMovies();
  const [hasSearched, setHasSearched] = useState<boolean>(false);
  const [searchQuery, setSearchQuery] = useState<string>('');
  const [pageNumber, setPageNumber] = useState<number>(1);
  const handleSearchInput = (text: string) => {
    setPageNumber(1);
    // Text is empty or search bar is cleared, no need to make requests
    if (!text) {
      setHasSearched(false);
      return;
    }
    setHasSearched(true);
    setSearchQuery(text);
    searchMoviesByText(text);
  };

  const handlePageChange = (selectedPageNumber: number) => {
    window.scrollTo({ top: 0, left: 0, behavior: 'smooth' });
    setPageNumber(selectedPageNumber);
    searchMoviesByText(searchQuery, selectedPageNumber);
  };

  return (
    <Container>
      <Box py={2}>
        <Input onChange={handleSearchInput} />
      </Box>
      {error && <ErrorLabel errorMessage={error} />}
      {isLoading && <CircularProgress size="large" color="secondary" />}
      {hasSearched === false ? (
        <IntroScreen />
      ) : data.length === 0 ? (
        <NoResultsSceen />
      ) : (
        <>
          <Grid container spacing={2}>
            {data.map((movie: Movie) => {
              return (
                <Grid key={movie.id} item xs={4}>
                  <MovieCard movie={movie} />
                </Grid>
              );
            })}
          </Grid>
          <Pagination
            currentPage={pageNumber}
            totalPages={totalPages}
            onChange={handlePageChange}
          />
        </>
      )}
    </Container>
  );
};

export default Homepage;
