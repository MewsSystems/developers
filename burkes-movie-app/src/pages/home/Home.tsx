import { debounce } from 'lodash';
import { useState } from 'react';
import { Link } from 'react-router-dom';

import { MovieCard } from '@/components/movieCard/MovieCard';
import { PageContainer } from '@/components/page/Page';
import { Pagination } from '@/components/pagination/Pagination';
import { useMoviesSearchQuery } from '@/queries/moviesQueries';

import css from './home.module.css';

export const Home = () => {
  const [search, setSearch] = useState('');
  const [currentPage, setCurrentPage] = useState(1);

  const handleSearch = (e: React.ChangeEvent<HTMLInputElement>) => {
    setSearch(e.target.value);
    setCurrentPage(1);
  };

  const debouncedSearch = debounce(handleSearch, 400);

  const {
    data,
    isLoading: isMovieListLoading,
    error,
  } = useMoviesSearchQuery(search, currentPage);

  const isSearchEmpty = !search;
  const isError = search && error;
  const isLoading = search && isMovieListLoading;
  const isFoundData = search && data?.results && data.results.length > 0;
  const isNoResults = search && data?.results && data.results.length === 0;

  return (
    <PageContainer>
      <div className={css.contentContainer}>
        <h1 className={css.title}>Mews Movie Search</h1>
        <input onChange={debouncedSearch} />
        <hr className={css.divider} />

        {isSearchEmpty && (
          <h1 className={css.title}>Please type something to begin search</h1>
        )}

        {isLoading && <h1 className={css.title}>Loading...</h1>}

        {isError && (
          <h1 className={css.title}>
            There seems to have been an error, the API might be down. Refer to
            the console for more details.
          </h1>
        )}

        {isFoundData && (
          <>
            <Pagination
              numberOfPages={data.total_pages}
              currentPage={currentPage}
              setCurrentPage={setCurrentPage}
            />

            <div className={css.movieCardsContainer}>
              {!isMovieListLoading && data && (
                <>
                  {data.results.map((movie) => (
                    <Link key={movie.id} to={`/${movie.id}`} state={movie}>
                      <MovieCard movie={movie} />
                    </Link>
                  ))}
                </>
              )}
            </div>
          </>
        )}

        {isNoResults && (
          <h1 className={css.title}>
            No results found, please alter your search
          </h1>
        )}
      </div>
    </PageContainer>
  );
};
