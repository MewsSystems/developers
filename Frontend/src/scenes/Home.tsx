import React, { useEffect, useState, useCallback } from 'react'
import { RouteComponentProps, navigate } from '@reach/router'
import { SceneMetaProps, SceneMeta } from 'components/SceneMeta'
import { Heading } from 'components/Heading'
import { useTranslation } from 'react-i18next'
import { useThunkDispatch } from 'hooks/useThunkDispatch'
import { hydrateTrendingMovies } from 'state/actions/movies'
import { useSelector } from 'react-redux'
import { State } from 'state/rootReducer'
import { differenceInDays } from 'date-fns'
import styled from 'styled-components'
import { MovieCard } from 'components/MovieCard'
import { Movie } from 'model/api/Movie'
import { Button } from 'components/Button'
import { useGetImagePath } from 'hooks/useGetImagePath'
import { ConfigurationBackdropSizesEnum } from 'model/api/ConfigurationBackdropSizesEnum'
import { Card as LoadingCard } from 'components/Loading/Card'
import { Spacing } from 'components/Spacing'
import { Content } from 'components/Content'

const Grid = styled.div`
  display: grid;
  grid-gap: 1rem;
  grid-template-columns: 1fr 1fr 1fr;
`

const { PUBLIC_URL } = process.env

export const Home: React.FC<RouteComponentProps<SceneMetaProps>> = ({
  title,
  description,
}) => {
  const { t } = useTranslation('home')
  const dispatch = useThunkDispatch()
  const [trendingMovies, setTrendingMovies] = useState<Movie[]>([])
  const [page, setPage] = useState(1)
  const {
    updatedAt,
    trending: { results, total_pages },
    loading,
  } = useSelector((state: State) => state.movies)
  const getImagePath = useGetImagePath(ConfigurationBackdropSizesEnum.W_780)
  const dayDifference = differenceInDays(new Date(), updatedAt)

  const handleLoadMore = useCallback(() => {
    setPage(page + 1)
  }, [page])

  const reduceTrendingMovies = useCallback(() => {
    return Object.keys(results).reduce<Movie[]>((acc, index) => {
      if (Number(index) <= page) {
        acc = [...acc, ...results[index]]
      }

      return acc
    }, [])
  }, [results, page])

  const handleCardClick = useCallback((id: number) => {
    navigate(`${PUBLIC_URL}/movie/${id}`)
  }, [])

  useEffect(() => {
    if (dayDifference > 7) {
      dispatch(hydrateTrendingMovies(true))
    } else {
      dispatch(hydrateTrendingMovies(false, page))
    }
  }, [dispatch, dayDifference, page])

  useEffect(() => {
    setTrendingMovies(reduceTrendingMovies())
  }, [page, results, reduceTrendingMovies])

  return (
    <>
      <SceneMeta title={title} description={description} />
      <Content>
        <Heading level={1}>{t('trending')}</Heading>
        {trendingMovies.length > 0 && (
          <>
            <Grid>
              {trendingMovies.map(
                ({ title, overview, backdrop_path, original_language, id }) => (
                  <MovieCard
                    key={id}
                    id={id}
                    title={title}
                    overview={overview}
                    background={getImagePath(backdrop_path)}
                    language={original_language}
                    onClick={handleCardClick}
                  />
                )
              )}
              {loading && (
                <>
                  <LoadingCard />
                  <LoadingCard />
                  <LoadingCard />
                  <LoadingCard />
                  <LoadingCard />
                </>
              )}
            </Grid>
            <Spacing outer="2rem 0 0">
              <Button
                onClick={handleLoadMore}
                disabled={page === total_pages || loading}
                fullWidth
              >
                {t('load-more')}
              </Button>
            </Spacing>
          </>
        )}
      </Content>
    </>
  )
}
