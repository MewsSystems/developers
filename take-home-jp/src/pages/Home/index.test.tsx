import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';
import Home from './';

it('renders correctly', () => {
  const tree = renderer.create(<Home />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('should show results after typing into search bar', () => {
  render(<Home />);
  const inputElement = screen.getByLabelText(/Search for a movie/i);
  fireEvent.change(inputElement, { target: { value: 'Godfather' } });
});
