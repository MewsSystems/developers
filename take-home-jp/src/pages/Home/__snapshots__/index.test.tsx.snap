// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
>
  <div
    className="MuiBox-root css-1sf3xto"
  >
    <div
      className="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
      style={
        Object {
          "backgroundColor": "#F5F5F5",
          "border": "1px solid #808080",
          "borderRadius": "5px",
          "display": "flex",
        }
      }
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-filled MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-filled css-e4w4as-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink={false}
        htmlFor="Movie Search"
        id="Movie Search-label"
      >
        Search for a movie
      </label>
      <div
        className="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-colorPrimary MuiInputBase-formControl css-batk84-MuiInputBase-root-MuiFilledInput-root"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoFocus={false}
          className="MuiInputBase-input MuiFilledInput-input css-10botns-MuiInputBase-input-MuiFilledInput-input"
          disabled={false}
          id="Movie Search"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="Search for a movie"
          required={false}
          type="text"
        />
      </div>
    </div>
  </div>
  <div
    className="MuiBox-root css-vxx7q6"
  >
    <h2>
      Welcome to Mews search
    </h2>
    <h4>
      Search for a movie above
    </h4>
  </div>
</div>
`;
